File: WEEK4/Junhyeok/1260.cs
using System;
using System.Linq;
using System.Collections.Generic;

class Solution {
    static int n, m, start;
    static bool[ , ] graph;
    static bool[] dfsVisited;
    static bool[] bfsVisited;
    
    static void Main() {
        InputData();
        Dfs(start);
        Console.WriteLine();
        Bfs();
    }
    
    static void InputData() {
        int[] input = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
        n = input[0]; m = input[1]; start = input[2];
        
        graph = new bool[n + 1, n + 1];
        dfsVisited = new bool[n + 1];
        bfsVisited = new bool[n + 1];
        for(int i = 0; i < m; i++) {
            int[] value = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
            graph[value[0], value[1]] = graph[value[0], value[1]] = true;
        }
    }

    static void Dfs(int start) {
        Console.Write($"{start} ");
        dfsVisited[start] = true;

        for(int i = 1; i <= n; i++) {
            if((graph[start, i] || graph[i, start]) && !dfsVisited[i]) Dfs(i);
        }
    }

    static void Bfs() {
        var queue = new Queue<int>();
        queue.Enqueue(start);
        bfsVisited[start] = true;

        while(queue.Count != 0) {
            int v = queue.Dequeue();
            Console.Write($"{v} ");

            for(int i = 1; i <= n; i++) {
                if((graph[v, i] || graph[i, v]) && !bfsVisited[i]) {
                    queue.Enqueue(i);
                    bfsVisited[i] = true;
                } 
            }
        }
    }
}

File: WEEK4/yonghyeon/1260.py
def dfs(t):
    dfs_list.append(t)   # 방문 노드
    v[t] = 1             # 표시

    for n in adj[t]:
        if not v[n]:
            dfs(n)

def bfs(c):
    queue = []

    queue.append(c)
    bfs_list.append(c)
    v[c] = 1

    while queue:
        t = queue.pop(0)
        for n in adj[t]:
            if not v[n]:    # 방문X 큐 삽입
                queue.append(n)
                bfs_list.append(n)
                v[n] = 1

n, m, start = map(int, input().split())
adj = [[] for _ in range(n+1)]
for _ in range(m):
    a, b = map(int, input().split())
    adj[a].append(b) # 양방향
    adj[b].append(a) 

for i in range(1, n+1):
    adj[i].sort() # 오름차순

v = [0]*(n+1)
dfs_list = []
dfs(start)

v = [0]*(n+1)
bfs_list = []
bfs(start)

print(*dfs_list)
print(*bfs_list)
